app/controllers/users/sessions_controller.rb

```diff
@@ -75,6 +81,38 @@ def process_locked_out_session
       warden.lock!
       flash[:error] = t('errors.sign_in.bad_password_limit')
       redirect_to root_url
+      :locked_out_session
+    end
+
+    def valid_captcha_result?
+      return true unless FeatureManagement.sign_in_recaptcha_enabled?
+      SignInRecaptchaForm.new(**recaptcha_form_args).submit(
+        email: auth_params[:email],
+        recaptcha_token: params.require(:user)[:recaptcha_token],
+        device_cookie: cookies[:device],
+      ).success?
+    end
+
+    def process_failed_captcha
+      flash[:error] = t('errors.messages.invalid_recaptcha_token')
+      warden.logout(:user)
+      warden.lock!
+      redirect_to root_url
+      :recaptcha
+    end
+
+    def recaptcha_form_args
+      args = { analytics: }
+      if IdentityConfig.store.phone_recaptcha_mock_validator
```

Comment:
Good catch! This was fixed in 112ea41c5d. It wasn't causing issues locally because of the short-circuit I added in d886823c9c, which was basically causing the mock validator not to be used (related to your comment in https://github.com/18F/identity-idp/pull/10772#discussion_r1644665974). I shifted this around in 474bc8ee75 to make sure reCAPTCHA validation is performed when using the mock validator..